ambry-store/src/main/java/com.github.ambry.store/LogSegment.java

```diff
@@ -24,10 +24,12 @@
 import java.io.IOException;
 import java.nio.ByteBuffer;
 import java.nio.channels.Channels;
+import java.nio.channels.ClosedChannelException;
 import java.nio.channels.FileChannel;
 import java.nio.channels.ReadableByteChannel;
 import java.util.concurrent.atomic.AtomicBoolean;
 import java.util.concurrent.atomic.AtomicLong;
+import net.smacke.jaydio.DirectRandomAccessFile;
```

Comment:
Is this a well-tested/accepted library? How did we come to know about this?

Reply:
https://github.com/smacke/jaydio

This library has tests and no known issues in the project page. But you are right, we are not sure how stable/safe this library is. 

So the plan is to have configuration for this feature and turn it on one machine to test it out(may be a few weeks) and then propagate to more hosts if we can see significant performance improvement. 

Meanwhile, I am looking at the code to understand more.

Reply:
Does it provide a lot of value over using JNA directly?

Reply:
It's a warp of JNA.

```
Jaydio uses JNA to accomplish all this parameter querying and other native magic.
```

Reply:
I looked at the same library about 2 years ago and I don't think anything has changed. The library isn't heavily used and I don't know if it is "standard". However it might be a useful substitute until we get around to implementing our own - what do you think? (esp since it is controlled by config)